### DISCUSSION TITLE
Hitting the MediaMarkt.de and Saturn.de API

### DISCUSSION THREAD

**Discussion by LINZO-AT** [jef/streetmerchant#1587] [Help]
Hello!

Due to many 403 Error (IP Bans) on the MediaSaturn-Sites I am trying to hit the Webshop APIs directly.
That's the API URL: 

https://www.mediamarkt.de/api/v1/graphql?operationName=GetProductAvailabilities&variables=%7B%22ids%22:%5B%222691244%22%5D%7D&extensions=%7B%22pwa%22%3A%7B%22salesLine%22%3A%22Saturn%22%2C%22country%22%3A%22DE%22%2C%22language%22%3A%22de%22%7D%2C%22persistedQuery%22%3A%7B%22version%22%3A1%2C%22sha256Hash%22%3A%22810286f7ae9368cb54ccd122b21e453bd00ed7dbf534b8259b8bed68f8da999f%22%7D%7D

Unforunately I am getting the following error:

> [9:53:24 AM] error :: ✖ [mediamarkt] asus 3070 strix - invalid json response body at https://www.mediamarkt.de/api/v1/graphql?operationName=GetProductAvailabilities&variables=%7B%22ids%22:%5B%222676607%22%5D%7D&extensions=%7B%22pwa%22%3A%7B%22salesLine%22%3A%22Saturn%22%2C%22country%22%3A%22DE%22%2C%22language%22%3A%22de%22%7D%2C%22persistedQuery%22%3A%7B%22version%22%3A1%2C%22sha256Hash%22%3A%22810286f7ae9368cb54ccd122b21e453bd00ed7dbf534b8259b8bed68f8da999f%22%7D%7D reason: Unexpected token < in JSON at position 0

According to Google, the error occurs because HTML code is being fetched and not JSON.
I would be very happy if someone could help me!

```
import {Store} from './store';
import fetch from 'node-fetch';

export const Mediamarkt: Store = {
	currency: '€',
	labels: {
		captcha: {
			container: 'p',
			text: ['Das ging uns leider zu schnell.']
		},
		maxPrice: {
			container: 'span[font-family="price"]',
			euroFormat: false
		},
		outOfStock: [
			{
				container: '#root',
				text: ['Dieser artikel ist aktuell nicht verfügbar.']
			},
			{
				container: '#root',
				text: ['Leider keine Lieferung möglich']
			},
			{
				container: '#root',
				text: ['Nicht verfügbar']
			},
			{
				container: '#root',
				text: ['Dieser Artikel ist dauerhaft ausverkauft']
			}
		]
	},
	links: [
		{
			brand: 'test:brand',
			model: 'test:model',
			series: 'test:series',
			url: 'https://www.mediamarkt.de/de/product/-2641856.html'
		},
		{
			brand: 'asus',
			model: 'strix',
			itemNumber: '2676607',
			series: '3070',
			url: 'https://www.mediamarkt.de/de/product/-2676607.html'
		},
		{
			brand: 'asus',
			model: 'strix oc',
			itemNumber: '2691244',
			series: '3070',
			url: 'https://www.mediamarkt.de/de/product/-2691244.html'
		}
	],
	name: 'mediamarkt',
	realTimeInventoryLookup: async (itemNumber: string) => {
		const request_url =
			'https://www.mediamarkt.de/api/v1/graphql?operationName=GetProductAvailabilities&variables=%7B%22ids%22:%5B%22' +
			itemNumber +
			'%22%5D%7D&extensions=%7B%22pwa%22%3A%7B%22salesLine%22%3A%22Saturn%22%2C%22country%22%3A%22DE%22%2C%22language%22%3A%22de%22%7D%2C%22persistedQuery%22%3A%7B%22version%22%3A1%2C%22sha256Hash%22%3A%22810286f7ae9368cb54ccd122b21e453bd00ed7dbf534b8259b8bed68f8da999f%22%7D%7D';
		const response = await fetch(request_url);
		const response_json = await response.json();
		return (
			response_json.data.availabilityType === "IN_WAREHOUSE"
		);
	}
};
```

Additional I have enabled the following part in lookup.ts:
```
	// Fixme: currently causing issues
	// Do API inventory validation in realtime (no cache) if available
if (
 	store.realTimeInventoryLookup !== undefined &&
 	link.itemNumber !== undefined
 ) {
 	return store.realTimeInventoryLookup(link.itemNumber);
 }
```

**Comment by marvelous052** [comment]
@LINZO-AT did you find a solution for the issue already ?

**Comment by johnconnor2020** [comment]
No
